src/control/cmd/agent/mgmt_rpc.go

```diff
@@ -63,14 +83,40 @@ func (mod *mgmtModule) ID() int32 {
 	return drpc.ModuleMgmt
 }
 
-func (mod *mgmtModule) handleGetAttachInfo(reqb []byte) ([]byte, error) {
-	if os.Getenv("DAOS_AGENT_DISABLE_CACHE") == "true" {
-		mod.cachedAttachInfo = false
-		mod.log.Debugf("GetAttachInfo agent caching has been disabled\n")
+// handleGetAttachInfo invokes the GetAttachInfo dRPC.  The agent determines the
+// NUMA node for the client process based on its PID.  Then based on the
+// server's provider, chooses a matching network interface and domain from the
+// client machine that has the same NUMA affinity.  It is considered an error if
+// the client application is bound to a NUMA node that does not have a network
+// device / provider combination with the same NUMA affinity.
+//
+// The client machine may have more than one matching network interface per
+// NUMA node.  In order to load balance the client application's use of multiple
+// network interfaces on a given NUMA node, a round robin resource allocation
+// scheme is used to choose the next device for that node.  See attachInfoCache
+// "loadBalance()" for details.
+//
+// The agent caches the local device data and all possible responses the first
+// time this dRPC is invoked. Subsequent calls receive the cached data.
+// The use of cached data may be disabled by exporting
+// "DAOS_AGENT_DISABLE_CACHE=true" in the environment running the daos_agent.
+func (mod *mgmtModule) handleGetAttachInfo(reqb []byte, pid int32) ([]byte, error) {
+	var err error
+	numaNode := defaultNumaNode
+
+	if mod.numaAware {
+		numaNode, err = netdetect.GetNUMASocketIDForPid(pid)
+		if err != nil {
+			return nil, err
+		}
 	}
 
-	if mod.cachedAttachInfo {
-		return mod.resmgmtpb, nil
+	if mod.aiCache.enabled.IsTrue() && mod.aiCache.initialized.IsTrue() {
+		resmgmtpb, err := mod.aiCache.getResponse(numaNode)
+		if err != nil {
+			return nil, err
+		}
+		return resmgmtpb, nil
```

Comment:
This is fine, but you could probably just `return mod.aiCache.getResponse(numaNode)` here, right?

Reply:
Yes - good idea.